### ISSUE TITLE
CommandBar A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#6390]
Tracking all of the issues brought up with the 09/14 accessibility pass on Command Bar  control
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B2ae54845-a1c2-4e8e-9d3c-d418976e5d30%7D&action=default&uid=%7B2AE54845-A1C2-4E8E-9D3C-D418976E5D30%7D&ListItemId=1371&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- [x] ~~Bug 1 : [Accessibility][Programmatic access]: Incorrect 'Aria-label' is defined for 'New’ CommandBar control.~~ Fixed in #8155.
- [x] ~~Bug 2 : [Accessibility][Programmatic access]: Elements must only use allowed ARIA attributes.~~ Doesn't repro anymore.
- [x] ~~Bug 3 : [Accessibility][Programmatic access]:Buttons must have discernible text.~~ Fixed in #8163.
- [x] ~Bug 4 : [Accessibility][Visual requirements]: 'Tooltip' is not defined for 'More option' control.~ Made separate issue to track this in #9873.
- [x] ~~Bug 5 : [Accessibility][Programmatic access]: Lists defined under 'menubar' do not have category/group name.~~ Duplicate of #6733.
- [x] ~~Bug 6 : [Accessibility][Programmatic access]: 'Aria-sort' property is not defined on 'Sort' button which will be conveying 'ascending/descending' order.~~ By design.
- [x] ~~Bug 7 : [Accessibility][Supporting the platform]:'Upload link' is not adopting high contrast black and white theme setting.~~ Doesn't repro anymore.
- [x] ~~Bug 8 : [Accessibility][Usable]:Scan mode focus is not visible on 'Grid view', 'More option' and 'Info button'.~~ Doesn't repro anymore.
- [x] ~~Bug 9 : [Accessibility][Visual Requirements]: Elements must have sufficient color contrast.~~ Fixed in #8566.
- [x] Bug 10 : [Accessibility][Suggestion][Keyboard navigation]: List items should only be navigable through up/down arrow keys. #10514
- [x] ~~Bug 11 : [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation.~~ Duplicate of bug 6 in #6356.
- [x] ~~Bug 12 : [Accessibility][Screen reader]: Narrator focus moves twice on the first element of context menu in scan mode.~~ Duplicate of bug 3 in #6360.
- [x] ~~Bug 13 : [Accessibility][Screen reader]: In Scan mode user is not able to close the 'Contextual menu' using 'Esc key'.~~ Duplicate of bug 7 in #6360.
- [x] ~~Bug 14 : [Accessibility][Screen reader]: Narrator focus moves onto hidden elements in scan mode in backward navigation.~~ Duplicate of bug 10 in #6360.
- [x] ~~Bug 15: [Accessibility][Commandbar]: Commandbar screen is not adopting to 400% scaling.~~ Duplicate of bug #9712

Component: CommandBar
@micahgodbolt

**Comment by micahgodbolt** [comment]
@lherrle can you explain your need? Tooltips are only used in CommandBar when a normal command is missing the button text. Are you asking for the overflow button to always have a tooltip?

@betrue-final-final - do you know if command bar should always have a tooltip? I'm assuming that the user would need to provide that string.

**Comment by lherrle** [comment]
@micahgodbolt I looked at the details on the sharepoint document -- the missing name for the "more options" button is what our bug is. We don't need an actual tooltip.

**Comment by micahgodbolt** [comment]
define "name"? Do you want text inline?

![image](https://user-images.githubusercontent.com/1434956/50243842-218d0580-0383-11e9-9242-cde630318aac.png)

Or do you want a title property?
![image](https://user-images.githubusercontent.com/1434956/50243893-497c6900-0383-11e9-8e49-179c5e9ec839.png)

Or do you just mean <button name="button"/>

**Comment by lherrle** [comment]
@micahgodbolt I believe the title property or aria-label could fix it; we had the same Keros result as 
![image](https://user-images.githubusercontent.com/2906990/50244297-767d4b80-0384-11e9-93e1-23196daabc69.png)

**Comment by micahgodbolt** [comment]
Fabric is language neutral, so aria-labels and such are up to the application to provide. 

The commandbar exposes the overflow icon props, so it is easy to include an aria-label prop to it

https://codepen.io/micahgodbolt/pen/VqmoRN

**Comment by khmakoto** [comment]
Crossing out __Bugs 12 and 13__ as they are being tracked by bugs 3 and 7 respectively in #6360.

**Comment by khmakoto** [comment]
Crossing out __Bug 11__ as it is being tracked by bug 6 in #6356.

**Comment by khmakoto** [comment]
Crossing out __Bug 2__ as it seems that this issue has been fixed and doesn't repro anymore.

**Comment by khmakoto** [comment]
Crossing out __Bug 8__ as it seems that this issue has been fixed and doesn't repro anymore.

**Comment by khmakoto** [comment]
__Bug 9__

Hi @TestAccessibilityGIT. When I run Keros (Version 1.200.27) on the webpage, the tool doesn't show me the color contrast issues that you have filed in the bug, even though I've double checked that the colors themselves have not changed.

Do you think this is an issue with Keros' version? If so, should I still try to fix the bug (would be difficult without the ability to check with Keros if the contrast is correct) or should I just cross out the bug?

Thanks!

**Comment by TestAccessibilityGIT** [comment]
Adding Amit to verify this defect.

Regards,
Arvind

**Comment by khmakoto** [comment]
Crossing out __Bug 6__ since it is asking us to define the `aria-sort` prop on the `Sort` button. However, the [`aria-sort` spec](https://www.w3.org/TR/wai-aria-1.1/#aria-sort) specifies that this property should only be applied to table headers or grid headers.

**Comment by khmakoto** [comment]
Crossing out **Bug 7** as it seems that this issue has been fixed and doesn't repro anymore.

**Comment by khmakoto** [comment]
Crossing out __Bug 5__ as it is being tracked in #6733.

**Comment by khmakoto** [comment]
Hi @TestAccessibilityGIT, any updates on the __Bug 9__ issue above?

**Comment by khmakoto** [comment]
Crossing out **Bug 9** as it doesn't repro with Keros anymore and there haven't been any updates on the verification of this bug. I'll uncross it if needed.

**Comment by v-doku** [comment]
@khmakoto , I verified the issue using Color contrast analyzer tool and it is still repro on our end. Please find the screenshot below.

![image](https://user-images.githubusercontent.com/47735112/54907746-42c63a80-4f0c-11e9-9ee7-e94cda741914.png)

**Comment by khmakoto** [comment]
Hi @v-doku, do you know where can I get this Color contrast analyzer tool?

**Comment by micahgodbolt** [comment]
btw, that red color is just a random color we grabbed to show how the buttons could be customized. We can easily use a different color

**Comment by sonimishra** [comment]
As checked , issue is fixed now..
In Narrator scan mode, focus does not moves to add image under Data Integration tab item
URL : https://preview.web.powerapps.com
Build version : 1809(OS build 18309.1000)​
Closing this bug

**Comment by v-doku** [comment]
@khmakoto you can find the color contrast analyzer tool [here.](url)

**Comment by micahgodbolt** [comment]
@khmakoto the command bar "more" button has the same issue as the breadcrumb. We need to figure out a way to address both (though the change probably comes from overflow set)

**Comment by khmakoto** [comment]
Commenting the same comment here as I did in #9688 

It seems we're doing very different things in different controls:
- We have `tooltipHostProps` in `Breadcrumb`.
- We also have `tooltipHostProps` in `CommandBar` (which creates a separate question about how to have multiple `Tooltip` props for different parts of a component).
- We use `title` in `Nav`, `SelectableOption` and `SpinButton`.
- We use the `label` in `GridCell` and `ColorPickerGridCell`.
- We use a render function in `DetailsColumn` and `DetailsHeader`.

**Comment by khmakoto** [comment]
I've made a separate issue to track bug 4 and similar issues in #9873.

**Comment by Kumar2608** [comment]
Bug 1 : [Accessibility][Programmatic access]: Incorrect 'Aria-label' is defined for 'New’ CommandBar control.
Correct "Aria-label"is defined for 'New' CommandBar control.
Issue is fixed. Verified the issue in below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 2 : [Accessibility][Programmatic access]: Elements must only use allowed ARIA attributes. Doesn't repro anymore.
Allowed attributes are used for the elements in the CommandBar screen.
Issue is fixed. Verified the issue in the below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 5 : [Accessibility][Programmatic access]: Lists defined under 'menubar' do not have category/group name. Duplicate of #6733.
Links defined under 'menubar" has category/group name.
Issue is fixed. Verified the issue in the below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 8 : [Accessibility][Usable]:Scan mode focus is not visible on 'Grid view', 'More option' and 'Info button'. Doesn't repro anymore.
Scan mode focus is visibile on 'Grid view', "More option', 'Info button'.
Issue is fixed. Verified the issue in the below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 9 : [Accessibility][Visual Requirements]: Elements must have sufficient color contrast. Fixed in #8566.
Luminosity is passing for the elements in CommandBar Screen.
Issue is fixed. Verified the issue in the below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 11 : [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation. Duplicate of bug 6 in #6356.
Screen scrolls down along with keyboard focus in back navigation.
Issue is fixed. Verified the issue in below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 12 : [Accessibility][Screen reader]: Narrator focus moves twice on the first element of context menu in scan mode. Duplicate of bug 3 in #6360.
Narrator focus moves only once to the first element of context menu in scan mode.
Issue is fixed. Verified the issue in below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

Bug 14 : [Accessibility][Screen reader]: Narrator focus moves onto hidden elements in scan mode in backward navigation. Duplicate of bug 10 in #6360.
Focus is not moving to the hidden elements in scan mode in backward navigation.
Issue is fixed. Verified the issue in below build:
OS: Version: 1903(OS Build 18956.1000)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/commandbar
Browser: Edge
Fabric site version id : 7.31.2

**Comment by kenotron** [comment]
@kumar2608 - would you mind searching for dupes before adding a new issue in this area?
This is actually a dupe of #9712.

**Comment by kenotron** [comment]
Opened a new bug to address the single issue left over.
